lib/argon.rb

```diff
@@ -159,17 +159,16 @@ class << self
             end
           end
 
-          if self.send(field) != from
-            if on_failed_transition
-              self.on_failed_transition(field: field, action: action, from: from, to: to)
-            end
-            raise Argon::InvalidTransitionError.new("Invalid state transition. #{ self.class.name }##{ self.send(self.class.primary_key.to_sym) } cannot perform '#{action}' on #{field}='#{from}'")
-          end
-
+          before_lock_state = self.send(field)
           begin
             self.with_lock do
```

Comment:
wow, for each state transition it locks the database record :-/. Not sure if this is required but If possible good to change it to `self.transaction do`.